### PR TITLE
Compile brutefource and knowhere core on Mac platform

### PR DESCRIPTION
**Pull request by jaime0815**

### CODE REVIEW

File: knowhere/knowhere/common/easyloggingpp/easylogging++.cc [hunk 1]
```diff
@@ -0,0 +1,3299 @@
+//
```
Comment (cydrain) [inline knowhere/knowhere/common/easyloggingpp/easylogging++.cc]: this file has already in "knowhere/thirdparty/easyloggingpp"

File: knowhere/knowhere/common/easyloggingpp/easylogging++.h [hunk 2]
```diff
@@ -0,0 +1,5193 @@
+//
```
Comment (cydrain) [inline knowhere/knowhere/common/easyloggingpp/easylogging++.h]: this file has already in "knowhere/thirdparty/easyloggingpp"

File: knowhere/knowhere/index/vector_index/faiss_common/AuxIndexStructures.cpp [hunk 3]
```diff
@@ -0,0 +1,326 @@
+/**
```
Comment (cydrain) [inline knowhere/knowhere/index/vector_index/faiss_common/AuxIndexStructures.cpp:1]: this file should not be copied out, knowhere does not depend on it

File: knowhere/knowhere/index/vector_index/faiss_common/AuxIndexStructures.h [hunk 4]
```diff
@@ -0,0 +1,258 @@
+/**
```
Comment (cydrain) [inline knowhere/knowhere/index/vector_index/faiss_common/AuxIndexStructures.h:1]: this file should not be copied out, knowhere does not depend on it

File: knowhere/knowhere/index/vector_index/faiss_common/FaissAssert.h [hunk 5]
```diff
@@ -0,0 +1,95 @@
+/**
```
Comment (cydrain) [inline knowhere/knowhere/index/vector_index/faiss_common/FaissAssert.h:1]: this file should not be copied out, knowhere does not depend on it

File: knowhere/knowhere/index/vector_index/faiss_common/FaissException.cpp [hunk 6]
```diff
@@ -0,0 +1,66 @@
+/**
```
Comment (cydrain) [inline knowhere/knowhere/index/vector_index/faiss_common/FaissException.cpp:1]: this file should not be copied out, knowhere does not depend on it

File: knowhere/knowhere/index/vector_index/faiss_common/FaissException.h [hunk 7]
```diff
@@ -0,0 +1,71 @@
+/**
```
Comment (cydrain) [inline knowhere/knowhere/index/vector_index/faiss_common/FaissException.h:1]: this file should not be copied out, knowhere does not depend on it

File: knowhere/knowhere/index/vector_index/faiss_common/AuxIndexStructures.h (hunk 4 above)
Comment (cydrain) [inline knowhere/knowhere/index/vector_index/faiss_common/AuxIndexStructures.h:1]: This file is only used by IDMAP::QueryByDistance() and BinaryIDMAP::QueryByDistance(),
and these 2 APIs have no corresponding SDK.
Suggest to "#if 0" these 2 APIs, and we can not copy AuxIndexStructure.h

File: CMakeLists.txt [hunk 8]
```diff
@@ -88,9 +105,17 @@ if (MILVUS_SUPPORT_SPTAG)
...
-add_subdirectory(knowhere/knowhere/utils)
-add_subdirectory(thirdparty)
+add_subdirectory(knowhere)
+if (MACOS)
+    include_directories(${INDEX_SOURCE_DIR}/thirdparty/easyloggingpp)
+    include_directories(${INDEX_SOURCE_DIR}/thirdparty/nolohmann)
+    add_subdirectory(selfdev/bruteforce)
+endif ()
+
+if (LINUX)
```
Comment (cydrain) [inline CMakeLists.txt:115]: use elif ()

File: knowhere/knowhere/common/Json.h [hunk 9]
```diff
@@ -11,7 +11,7 @@
 
 #pragma once
 
-#include "nlohmann/json.hpp"
+#include "../../../thirdparty/nlohmann/json.hpp"
```
Comment (cydrain) [inline knowhere/knowhere/common/Json.h:14]: don't use relative path

File: knowhere/knowhere/common/Log.h [hunk 10]
```diff
@@ -12,8 +12,7 @@
 #pragma once
 
 #include <string>
-
-#include "easyloggingpp/easylogging++.h"
+#include "../../../thirdparty/easyloggingpp/easylogging++.h"
```
Comment (cydrain) [inline knowhere/knowhere/common/Log.h:15]: don't use relative path

File: knowhere/knowhere/index/vector_index/faiss_common/MetricType.h [hunk 11]
```diff
@@ -0,0 +1,41 @@
+/**
```
Comment (cydrain) [inline knowhere/knowhere/index/vector_index/faiss_common/MetricType.h:1]: don't create this new directory "faiss_common", put this file "MetricType.h" under knowhere/utils

File: knowhere/knowhere/index/vector_index/helpers/DynamicResultSet.cpp [hunk 12]
```diff
@@ -10,11 +10,10 @@
 // or implied. See the License for the specific language governing permissions and limitations under the License.
 
 #include <cstring>
-#include <iostream>
 #include <string>
 #include <utility>
 
-#include "faiss/impl/AuxIndexStructures.h"
```
Comment (cydrain) [inline knowhere/knowhere/index/vector_index/helpers/DynamicResultSet.cpp:17]: it's wrong to change this header file path.
and this header file should be embraced by "#ifdef __linux__"

File: CMakeLists.txt (hunk 8 above)
Comment (jaime0815) [reply CMakeLists.txt:113]: already use elif() in 40 lines,  so if() more clear,  elif() should be used with else() together

**Pull request merged**